x-pack/filebeat/input/httpjson/input.go

```diff
@@ -0,0 +1,337 @@
+				if err != nil {
+					in.log.Info("Successfully processed HTTP request. Pagination finished.")
+					return nil
+				}
+				if in.config.Pagination.RequestField != "" {
+					ri.ContentMap.Put(in.config.Pagination.RequestField, v)
+					if in.config.Pagination.URL != "" {
+						ri.URL = in.config.Pagination.URL
+					}
+				} else {
+					switch v.(type) {
+					case string:
+						ri.URL = v.(string)
+					default:
+						return errors.New("pagination ID is not string, which is required for URL - ")
+					}
+				}
+				if in.config.Pagination.ExtraBodyContent != nil {
+					ri.ContentMap.Update(common.MapStr(in.config.Pagination.ExtraBodyContent))
+				}
+				req, err = in.createHTTPRequest(ctx, ri)
+				if err != nil {
+					in.log.Error(err)
+					return err
+				}
+				continue
+			}
+			return nil
+		default:
+			return errors.New("response is not valid JSON - ")
+		}
+	}
+}
+
+func (in *httpjsonInput) run() error {
+	ctx, cancel := context.WithCancel(in.workerCtx)
+	defer cancel()
+
+	tlsConfig, err := tlscommon.LoadTLSConfig(in.config.TLS)
+	if err != nil {
+		return err
+	}
+
+	var dialer, tlsDialer transport.Dialer
+
+	dialer = transport.NetDialer(in.config.HTTPClientTimeout)
+	tlsDialer, err = transport.TLSDialer(dialer, tlsConfig, in.config.HTTPClientTimeout)
+	if err != nil {
+		return err
+	}
+
+	// Make transport client
+	var client *http.Client
+	client = &http.Client{
+		Transport: &http.Transport{
+			Dial:              dialer.Dial,
+			DialTLS:           tlsDialer.Dial,
+			DisableKeepAlives: true,
+		},
+		Timeout: in.config.HTTPClientTimeout,
+	}
+
+	// if in.config.ServerName == "" {
+	// 	in.log.Info("ServerName is empty, hence TLS will not be used.")
+	// 	client = &http.Client{
+	// 		Timeout:   time.Second * time.Duration(in.config.HTTPClientTimeout),
+	// 		Transport: &http.Transport{DisableKeepAlives: true},
+	// 	}
+	// } else {
+	// 	in.log.Info("ServerName is " + in.config.ServerName + ", and TLS  will be used.")
+	// 	client = &http.Client{
+	// 		Timeout: time.Second * time.Duration(in.config.HTTPClientTimeout),
+	// 		Transport: &http.Transport{
+	// 			TLSClientConfig: &tls.Config{
+	// 				ServerName: in.config.ServerName,
+	// 				// InsecureSkipVerify: true,
+	// 			},
+	// 			DisableKeepAlives: true,
+	// 		},
+	// 	}
+	// }
+	ri := &requestInfo{
+		URL:        in.URL,
+		ContentMap: common.MapStr{},
+	}
+	if in.config.HTTPMethod == "POST" && in.config.HTTPRequestBody != nil {
+		ri.ContentMap.Update(common.MapStr(in.config.HTTPRequestBody))
+	}
+	req, err := in.createHTTPRequest(ctx, ri)
+	if err != nil {
+		return err
+	}
+	err = in.processHTTPRequest(ctx, client, req, ri)
+	if err == nil && in.Interval > 0 {
+		ticker := time.NewTicker(time.Duration(in.Interval) * time.Second)
+		for {
+			select {
+			case <-ctx.Done():
+				in.log.Info("Context done.")
+				return nil
+			case <-ticker.C:
+				err = in.processHTTPRequest(ctx, client, req, ri)
```

Comment:
IIUC this error is never returned. The `for` loop only stops on the `return nil`.

Reply:
I just noticed this too.